pkg/pow/pow.go

```diff
@@ -0,0 +1,113 @@
+package pow
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/iotaledger/iota.go/pow"
+	"github.com/iotaledger/iota.go/trinary"
+	powsrvio "gitlab.com/powsrv.io/go/client"
+)
+
+const powsrvReinitCooldown = 30 * time.Second
+
+// Handler struct
+type Handler struct {
+	sync.RWMutex
+
+	powsrvClient      *powsrvio.PowClient
+	localPoWFunc      pow.ProofOfWorkFunc
+	powType           string
+	lastInitTimestamp int64
+}
+
+// tryReinitPowsrvWithoutLocking tries to reinit powsrv.io if the connection got lost.
+// the lock needs to be handled by the caller
+func (h *Handler) tryReinitPowsrvWithoutLocking() {
+	if h.powsrvClient == nil {
+		return
+	}
+
+	if h.lastInitTimestamp+powsrvReinitCooldown.Nanoseconds() <= time.Now().UnixNano() {
+		h.powsrvClient.Close()
+		h.powsrvClient.Init()
+		h.lastInitTimestamp = time.Now().UnixNano()
+	}
+}
+
+// NewHandler creates a new PoW handler instance
+func NewHandler() (*Handler, error) {
+
+	var localPoWFunc pow.ProofOfWorkFunc
```

Comment:
Create a pow plugin and init the Handler globally with:

```go
var (
	PLUGIN      = node.NewPlugin("PoW", node.Enabled, configure, run)
	log         *logger.Logger
	handler     *pow.Handler
	handlerOnce sync.Once
)


// Handler gets the pow Handler instance.
func Handler() *pow.Handler {
	handlerOnce.Do(func() {
		// init the pow handler with all possible settings
	})
	return handler
}
```

Then you can call `pow.Handler()` from all plugins that use it, and you can connect the shutdown event cleanly with its own priority.